u2f-host/u2fmisc.c

```diff
@@ -222,37 +222,45 @@ u2fh_sendrecv (u2fh_devs * devs, unsigned index, uint8_t cmd,
     int timeout = HID_TIMEOUT;
     int rc = 0;
 
-    while (rc == 0)
+    do
       {
-	if (debug)
+	timeout = HID_TIMEOUT;
+	rc = 0;
+
+	while (rc == 0)
 	  {
-	    fprintf (stderr, "now trying with timeout %d\n", timeout);
+	    if (debug)
+	      {
+		fprintf (stderr, "now trying with timeout %d\n", timeout);
+	      }
+	    rc = hid_read_timeout (dev->devh, data, len, timeout);
+	    timeout *= 2;
+	    if (timeout > HID_MAX_TIMEOUT)
+	      {
+		rc = -2;
+		break;
+	      }
 	  }
-	rc = hid_read_timeout (dev->devh, data, len, timeout);
-	timeout *= 2;
-	if (timeout > HID_MAX_TIMEOUT)
+	sequence = 0;
```

Comment:
sequence should probably be reset before the loop, not inside since it's not used inside.